samtranslator/plugins/api/implicit_api_plugin.py

```diff
@@ -59,15 +61,20 @@ def on_before_transform_template(self, template_dict):
 
         template = SamTemplate(template_dict)
 
+        # Adjust self.implicit_api_logical_id if the template already contains a resource with the same logical ID
+        # Appending a number (2) to it. If it is still not unique, increment the number until the logical ID is unique.
+        resource_logical_ids = set(template.resources.keys())
+        if self.implicit_api_logical_id in resource_logical_ids:
+            suffix = 2
+            while (self.implicit_api_logical_id + str(suffix)) in resource_logical_ids:
+                suffix = suffix + 1
+            self.implicit_api_logical_id = self.implicit_api_logical_id + str(suffix)
+            LOG.debug('Adjust implicit API logical ID to "%s"', self.implicit_api_logical_id)
```

Comment:
I assume `ServerlessRestApi` (and other reserved words) isn't hardcoded elsewhere that could break a reference?

Could also make this a `disambiguate(name: str, names: list[str]) -> str` if this is expected to be a common pattern.

Reply:
yes, it is not hardcoded except in tests. 

> disambiguate(name: str, names: list[str]) -> str

sounds like a good idea